### DISCUSSION TITLE
Not using http transport for `personal_sign` method.

### DISCUSSION THREAD

**Discussion by pepoospina** [wevm/viem#1538] [Question]
I'm using a wallet client with a programmed private key to test my app on a testnet. 

I need to use Alchemy instead of a local testnet and, so, this is how I configure it:

```ts
  const account = mnemonicToAccount(mnemonic, { accountIndex: ix });

  const client = createWalletClient({
    account,
    chain,
    transport: http(`https://${ALCHEMY_SUBDOMAIN}.g.alchemy.com/v2/${ALCHEMY_KEY}`),
  });

  const signer = new WalletClientSigner(client, 'json-rpc');
```

I now want to sign a message with `signer.signMessage()` but it forwards the request to Alchemy and throws the error

```
Request body: {"method":"personal_sign","params":["0x5072657061726520616e6f6e796d6f7573206964656e74697479","0xFfEB3c7bE84527a44349aFD421dC5a1834DBAa1D"]}

Details: {"code":-32600,"message":"Unsupported method: personal_sign
```

I would like this method (personal_sign), to be handled by the connected account instead.

How would you achieve this? Should I build a custom transport?

**Comment by jxom** [comment]
What is `WalletClientSigner`? The `signMessage` Action in Viem should be able to sign a message locally without broadcasting over JSON-RPC.

**Reply by pepoospina** [reply]
Oh, sorry, I was confused. `WalletClientSigner` is actually an account abstraction wrapper, so need to check that. I'm forgetting what my own code is doing...

This is the wrapper btw: https://accountkit.alchemy.com/packages/aa-core/signers/wallet-client.html

**Reply by jxom** [reply]
Interesting. You could probably use `signMessage` from the `account` directly (`account.signMessage({ message: 'blah' })`)

**Reply by jxom** [reply]
https://viem.sh/docs/accounts/signMessage.html
